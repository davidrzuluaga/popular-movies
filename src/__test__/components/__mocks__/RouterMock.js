import React from 'react';
import { Router } from 'react-router-dom';
import { createBrowserHistory } from 'history';

const history = createBrowserHistory();

const RouterMock = props => (
    <Router history={history}>
      {props.children}
    </Router>
);

export default RouterMock;
