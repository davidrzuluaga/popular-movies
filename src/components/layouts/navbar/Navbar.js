import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = props => {
  return (
    <div className="navbar">
      <h1>Welcome to the Popular Movies List</h1>
      <div className="buttons">
        <Link to={`/`}>
          <button>Top Movies</button>
        </Link>
        <Link to={`/favorites`}>
          <button>
            My favorites
          </button>
        </Link>
      </div>
    </div>
  );
};

export default Navbar;
