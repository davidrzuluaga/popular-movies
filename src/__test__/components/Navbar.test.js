import React from 'react';
import { mount } from 'enzyme';
import Navbar from '../../components/layouts/navbar/Navbar';
import RouterMock from './__mocks__/RouterMock';

describe('<Navbar />', () => {
  const navbar = mount(
    <RouterMock>
      <Navbar />
    </RouterMock>
    );

  test('Render del componente Navbar', () => {
    expect(navbar.length).toEqual(1);
  });
  test('Render del titulo', () => {
    expect(navbar.find('h1').text()).toEqual('Welcome to the Popular Movies List');
  });
});