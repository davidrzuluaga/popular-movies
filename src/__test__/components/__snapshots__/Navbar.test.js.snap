// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Snapshot Comprobar la UI del componente Navbar 1`] = `
<div
  className="navbar"
>
  <h1>
    Welcome to the Popular Movies List
  </h1>
  <div
    className="buttons"
  >
    <button
      onClick={[Function]}
    >
      Top Movies
    </button>
    <button
      onClick={[Function]}
    >
      My favorites
    </button>
  </div>
</div>
`;
